File: domain/src/main/java/com/imadji/arch/domain/usecase/GetPopularMovies.java
package com.imadji.arch.domain.usecase;

import com.imadji.arch.domain.model.Movie;
import com.imadji.arch.domain.repository.MovieRepository;
import com.imadji.arch.domain.usecase.type.SingleUseCase;

import java.util.List;

import io.reactivex.Single;

/**
 * Created by imadji on 06/20/2019.
 */

public class GetPopularMovies implements SingleUseCase<List<Movie>> {
    private final MovieRepository movieRepository;

    public GetPopularMovies(MovieRepository movieRepository) {
        this.movieRepository = movieRepository;
    }

    @Override
    public Single<List<Movie>> execute() {
        return movieRepository.getPopularMovies();
    }
}

File: domain/src/main/java/com/imadji/arch/domain/usecase/GetPopularMovies.kt
package com.imadji.arch.domain.usecase

import com.imadji.arch.domain.entity.Movie
import com.imadji.arch.domain.repository.MovieRepository
import com.imadji.arch.domain.usecase.type.SingleUseCase

import io.reactivex.Single
import javax.inject.Inject

class GetPopularMovies @Inject constructor(
        private val movieRepository: MovieRepository
) : SingleUseCase<List<Movie>> {

    override fun execute(): Single<List<Movie>> {
        return movieRepository.getPopularMovies()
    }
}